pkg/extract/compression.go

```diff
@@ -0,0 +1,137 @@
+package extract
+
+import (
+	"compress/bzip2"
+	"compress/gzip"
+	"compress/lzw"
+	"encoding/binary"
+	"io"
+
+	"github.com/pierrec/lz4"
+	"github.com/ulikunitz/xz"
+
+	"github.com/replicate/pget/pkg/logging"
+)
+
+const (
+	peekSize = 8
+
+	gzipMagic = 0x1F8B
```

Comment:
did you consider making these byte slices rather than integers? Then you don't have to do mucking around with binary.BigEndian or padding, you can just straight up use bytes.HasPrefix. eg:
```suggestion
	gzipMagic = []byte{0x1F,0x8B}
```

Reply:
I remember why! Because I wanted to use constants for these vs vars -- which is silly looking back on it now.